src/test/integration/DesignSurface/DesignSurfaceExt/Framework/Analyzer.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Diagnostics.CodeAnalysis;
+
+// C# only looks at the type name. To allow null analysis to work on .NET Framework, we need to copy the attributes code.
+[AttributeUsage(AttributeTargets.Parameter, Inherited = false)]
+public sealed class NotNullWhenAttribute : Attribute
```

Comment:
```suggestion
internal sealed class NotNullWhenAttribute : Attribute
```
It does not have to be public